src/StripeTests/Infrastructure/Public/StripeClientTest.cs

```diff
@@ -0,0 +1,112 @@
+namespace StripeTests
+{
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Stripe;
+    using Xunit;
+
+    public class StripeClientTest : BaseStripeTest
+    {
+        private readonly DummyHttpClient httpClient;
+        private readonly StripeClient stripeClient;
+        private readonly BaseOptions options;
+        private readonly RequestOptions requestOptions;
+
+        public StripeClientTest()
+        {
+            this.httpClient = new DummyHttpClient();
+            this.stripeClient = new StripeClient(this.httpClient);
+            this.options = new ChargeCreateOptions
+            {
+                Amount = 123,
+                Currency = "usd",
+                SourceId = "tok_visa",
+            };
+            this.requestOptions = new RequestOptions();
+        }
+
+        [Fact]
+        public async Task RequestAsync_OkResponse()
+        {
+            var response = new StripeResponse(HttpStatusCode.OK, null, "{\"id\": \"ch_123\"}");
+            this.httpClient.Response = response;
+
+            var charge = await this.stripeClient.RequestAsync<Charge>(
+                HttpMethod.Post,
+                "/v1/charges",
+                this.options,
+                this.requestOptions);
+
+            Assert.NotNull(charge);
+            Assert.Equal("ch_123", charge.Id);
+            Assert.Equal(response, charge.StripeResponse);
+        }
```

Comment:
Just wanted to echo again that I love how short + succinct these sorts of tests come out! Very readable.